webui/react/src/shared/components/ActionDropdown/ActionDropdown.test.tsx

```diff
@@ -0,0 +1,90 @@
+import { render, screen, waitFor } from '@testing-library/react';
+import userEvent, { PointerEventsCheckLevel } from '@testing-library/user-event';
+import React from 'react';
+
+import ActionDropdown from 'shared/components/ActionDropdown/ActionDropdown';
+
+const user = userEvent.setup({ pointerEventsCheck: PointerEventsCheckLevel.Never });
+
+const ACTION_ONE_TEXT = 'Action One';
+const ACTION_TWO_TEXT = 'Action Two';
+
+enum TestAction {
+  ActionOne = 'Action One',
+  ActionTwo = 'Action Two',
+}
+
+const handleActionOne = jest.fn();
+const handleActionTwo = jest.fn();
+
+const DropDownContainer = () => {
+
+  const dropDownOnTrigger = () => {
+    return {
+      [TestAction.ActionOne]: () => handleActionOne(),
+      [TestAction.ActionTwo]: () => handleActionTwo(),
+    };
+  };
+
+  return (
+    <ActionDropdown<TestAction>
+      actionOrder={[
+        TestAction.ActionOne,
+        TestAction.ActionTwo,
+      ]}
+      id={'test-id'}
+      kind="test"
+      onError={() => { return; }}
+      onTrigger={dropDownOnTrigger()}
+    />
+  );
+
+};
+
+const setup = () => {
+  const view = render(<DropDownContainer />);
+  return { view };
+};
+
+describe('ActionDropdown', () => {
+  setup();
+
+  it('should display trigger button', async () => {
+
+    await waitFor(() => {
+      expect(screen.getByRole('button')).toBeInTheDocument();
+    });
+
+  });
+
+  it('should display actions', async () => {
+    setup();
+
+    user.click(screen.getByRole('button'));
+
+    await waitFor(() => {
+      expect(screen.getByText(ACTION_ONE_TEXT)).toBeInTheDocument();
+    });
+
+    await waitFor(() => {
+      expect(screen.getByText(ACTION_TWO_TEXT)).toBeInTheDocument();
+    });
+
+  });
+
+  it('should call dropdown option one function', async () => {
+    setup();
```

Comment:
Adding `expect(handleActionOne).not.toHaveBeenCalled();` before user.click would be nice to test functions

Reply:
Agreed!